docroot/themes/custom/dct_theme/dct_theme.theme

```diff
@@ -13,3 +13,14 @@ function dct_theme_preprocess_page(&$variables) {
     $variables['#attached']['library'][] = 'dct_theme/parallax';
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK() for image slider paragraph.
```

Comment:
Preprocess for paragraph templates, you can add actions for other types of paragraphs.